### ISSUE TITLE
Problem with Forestry Build

### ISSUE THREAD

**Issue by RassanGrant** [Norstone/norstone.github.io#306]
Hey @vleong2332 - Could you please have a look at this Wercker Pipeline and let us know what's wrong and what needs to be fixed? 

https://github.com/Norstone/norstone.github.io/commit/2017d97ca5ffd59735a2239fd0c2f5f63bf58115

Slack says this is the issue: "build for norstone.github.io by rassan has failed on branch develop at step: Compile static assets" but Wercker says it's related to a "slack-notifier failure."

We just need to make sure Kevin can make edits via Forestry whenever he needs to. Let us know what's up and what you'd need timewise to fix it. Thanks!

**Comment by vleong2332** [comment]
@RassanGrant From what I see, the failure is caused by invalid `date` and `layout` on that commit.
https://github.com/Norstone/norstone.github.io/commit/2017d97ca5ffd59735a2239fd0c2f5f63bf58115#diff-bf56f944a1f49c739e7fb7f1a7b8997fR37

This sounds like an automatically generated commit. We can manually remove the offending lines and it will fix it for this time. I'm afraid this could happen again if we didn't know what the real problem is. Since I don't think I have access to Forestry, I would try to see if @kevinnorstone did anything different in there that may have caused those 2 lines to be generated with an empty value (`''`).

Let me know if I can be of more help.

**Comment by vleong2332** [comment]
@RassanGrant After looking at it some more, here's my conclusion.

TL; DR; What failed the build is an empty `date` on the `xlx-rock` page. The reason the date was empty was because it was the first time it was edited through Forestry.

Here's the history: https://github.com/Norstone/norstone.github.io/commits/develop/_products/xlx-rock/xlx-rock.html

History summary:
1. The file was created by hand some time ago (without `date` and `layout`)
2. It had been edited by hand for some time.
3. On Oct 28, it was edited through Forestry.
4. Forestry's form has `date` and `layout` fields that were initialized as empty (because of number 1).
5. When the edits were saved in Forestry, the `date` and `layout` were generated as empty on Github.
6. The empty `date` broke the build.

Fix proposal:
To avoid similar problem in the future, I would make the `date` field on Forestry as required.

Let me know if you need my help to do that.

**Comment by vleong2332** [comment]
Here's the summary of the solution:
- Forestry will now prevent you from saving an update if `date` and/or `layout` is empty.
- All pages that can be edited from Forestry now has `date` and `layout` values, which means Kevin won't be prevented from making an update through Forestry.
- The project looked and built fine, but if you see anything weird like that again, give me a holla.
